alacritty/src/macos/proc.rs

```diff
@@ -0,0 +1,104 @@
+use std::ffi::OsStr;
+use std::io;
+use std::mem;
+use std::os::raw;
+use std::os::unix::ffi::OsStrExt;
+use std::path::PathBuf;
+use std::slice;
+use memchr::memchr;
+
+pub fn cwd(pid: raw::c_int) -> io::Result<PathBuf> {
+    let flavor = PROC_PIDVNODEPATHINFO as raw::c_int;
+    let mut info = mem::MaybeUninit::<proc_vnodepathinfo>::uninit();
+    let size = mem::size_of::<proc_vnodepathinfo>() as raw::c_int;
+    let result = unsafe {
+        proc_pidinfo(pid, flavor, 0, info.as_mut_ptr() as *mut raw::c_void, size)
+    };
+    match result {
+        c if c <= 0    => return Err(io::Error::last_os_error()),
+        s if s != size => return Err(io::Error::new(io::ErrorKind::Other, "invalid return size")),
+        _ => {}
+    }
+    let raw_path = unsafe {
+        let info = info.assume_init();
+        slice::from_raw_parts(info.pvi_cdir.vip_path.as_ptr() as *const u8,
+                              info.pvi_cdir.vip_path.len())
+    };
+    let null_idx = memchr(0x00, &raw_path).unwrap_or(0);
```

Comment:
```suggestion
    let null_idx = memchr(0, &raw_path).unwrap_or(0);
```

Reply:
How about `'\0'`?

Reply:
I'd be fine with '\0', that would make it pretty clear what's going on.